automationexcersise/conftest.py

```diff
@@ -0,0 +1,142 @@
+import pytest
+import json
+from selenium import webdriver
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support.wait import WebDriverWait
+from selenium.webdriver.support import expected_conditions as EC
+import time
+
+@pytest.fixture()
+def get_driver(get_settings):
+    driver = webdriver.Firefox()
+    driver.implicitly_wait(get_settings["wait_duration"])
+    return driver
+    # yield driver
+    # driver.quit()
+
+
+@pytest.fixture()
+def get_settings():
+    with open("config.json") as f:
+        data = json.load(f)
+    return data["settings"]
+
+@pytest.fixture()
+def get_xpaths():
+    with open("config.json") as f:
+        data = json.load(f)
+    return data["xpaths"]
+
+
+@pytest.fixture()
+def get_signup(get_xpaths, get_driver, get_settings):
+    driver = get_driver
+    driver.get(get_settings["base_url"])
+
+    assert WebDriverWait(driver, 5).until(EC.presence_of_element_located((By.XPATH, get_xpaths["home"]["home_button"])))
+    login_button = driver.find_element(By.XPATH, get_xpaths["login"]["login_subpage"])
+    login_button.click()
+
+    assert WebDriverWait(driver, 5).until(EC.presence_of_element_located((By.XPATH, "//div[@class='signup-form']")))
+    signup_name = driver.find_element(By.XPATH, "//input[@data-qa='signup-name']")
+    signup_email = driver.find_element(By.XPATH, "//input[@data-qa='signup-email']")
+    signup_button = driver.find_element(By.XPATH, "//button[@data-qa='signup-button']")
+
+    signup_name.send_keys("John Smith")
+    signup_email.send_keys(get_settings["email"])
+    signup_button.click()
+
+    assert WebDriverWait(driver, 5).until(
+        EC.presence_of_element_located((By.XPATH, "//*[contains(text(), 'Enter Account "
+                                                  "Information')]")))
+    title = driver.find_element(By.XPATH, "//input[@id='id_gender1']")
+    password = driver.find_element(By.XPATH, "//input[@id='password']")
+    days = driver.find_element(By.XPATH, "//select[@id='days']/option[@value='1']")
+    months = driver.find_element(By.XPATH, "//select[@id='months']/option[@value='1']")
+    years = driver.find_element(By.XPATH, "//select[@id='years']/option[@value='2000']")
+    newsletter = driver.find_element(By.XPATH, "//input[@id='newsletter']")
+    optin = driver.find_element(By.XPATH, "//input[@id='optin']")
```

Comment:
same here. This will probably generate some stale elements

Reply:
Fixed